frontend/pages/settings/personal-access-tokens/new.tsx

```diff
@@ -18,79 +18,72 @@ import {
 } from '@mui/material'
 import { useTheme } from '@mui/material/styles'
 import { EntrySearchResult, useListModels } from 'actions/model'
+import { useGetUiConfig } from 'actions/uiConfig'
 import { postUserToken } from 'actions/user'
 import { ChangeEvent, SyntheticEvent, useCallback, useMemo, useState } from 'react'
+import Loading from 'src/common/Loading'
 import Title from 'src/common/Title'
 import Link from 'src/Link'
 import MessageAlert from 'src/MessageAlert'
 import TokenDialog from 'src/settings/authentication/TokenDialog'
-import { TokenAction, TokenActionKeys, TokenActionKind, TokenInterface, TokenScope } from 'types/types'
+import { TokenActionKind, TokenInterface, TokenScope } from 'types/types'
 import { getErrorMessage } from 'utils/fetcher'
 import { plural } from 'utils/stringUtils'
 
-const [TokenReadAction, TokenWriteAction] = Object.values(TokenAction).reduce<Record<string, TokenActionKeys>[]>(
-  ([readActions, writeActions], action) => {
-    let groupedActions = [readActions, writeActions]
-    const [name, kind] = action.split(':')
-
-    if (kind === TokenActionKind.READ) {
-      groupedActions = [{ ...readActions, [name]: action }, writeActions]
-    }
-    if (kind === TokenActionKind.WRITE) {
-      groupedActions = [readActions, { ...writeActions, [name]: action }]
-    }
-
-    return groupedActions
-  },
-  [{}, {}],
-)
-
-const isReadAction = (action: TokenActionKeys) => {
-  return Object.values(TokenReadAction).includes(action)
-}
-
-const isWriteAction = (action: TokenActionKeys) => {
-  return Object.values(TokenWriteAction).includes(action)
-}
+export default function NewToken() {
+  const { models, isModelsLoading, isModelsError } = useListModels('model')
+  const { uiConfig, isUiConfigLoading, isUiConfigError } = useGetUiConfig()
 
-const getActionName = (action: TokenActionKeys) => {
-  const [name, _kind] = action.split(':')
-  return name
-}
+  const tokenActions = uiConfig?.tokenActions || {}
 
-const actionOptions = Object.values(TokenAction)
+  const actionOptions = Object.keys(tokenActions)
```

Comment:
Minor, but as `uiConfig.tokenActions` is a decent size object I think it's worth `useMemo`ing this. You'll also have to useMemo `tokenActions` too to make that a stable dependency.